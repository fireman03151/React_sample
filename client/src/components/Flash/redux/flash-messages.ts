export enum FlashMessages {
  AccountDeleted = 'flash.account-deleted',
  AddNameSuccess = 'flash.add-name',
  AlreadyClaimed = 'flash.already-claimed',
  CertClaimSuccess = 'flash.cert-claim-success',
  CertificateMissing = 'flash.certificate-missing',
  CertsPrivate = 'flash.certs-private',
  ChallengeSaveTooBig = 'flash.challenge-save-too-big',
  ChallengeSubmitTooBig = 'flash.challenge-submit-too-big',
  CodeSaved = 'flash.code-saved',
  CodeSaveError = 'flash.code-save-error',
  CodeSaveLess = 'flash.code-save-less',
  CompleteProjectFirst = 'flash.complete-project-first',
  DeleteTokenErr = 'flash.delete-token-err',
  EmailValid = 'flash.email-valid',
  HonestFirst = 'flash.honest-first',
  IncompleteSteps = 'flash.incomplete-steps',
  LocalCodeSaved = 'flash.local-code-saved',
  LocalCodeSaveError = 'flash.local-code-save-error',
  NameNeeded = 'flash.name-needed',
  None = '',
  NotEligible = 'flash.not-eligible',
  NotHonest = 'flash.not-honest',
  NotRight = 'flash.not-right',
  ProfilePrivate = 'flash.profile-private',
  ProgressReset = 'flash.progress-reset',
  ProvideUsername = 'flash.provide-username',
  ReallyWeird = 'flash.really-weird',
  ReportSent = 'flash.report-sent',
  SigninSuccess = 'flash.signin-success',
  StartProjectErr = 'flash.start-project-err',
  TimelinePrivate = 'flash.timeline-private',
  TokenDeleted = 'flash.token-deleted',
  UpdatedAboutMe = 'flash.updated-about-me',
  UpdatedSocials = 'flash.updated-socials',
  UpdatedSound = 'flash.updated-sound',
  UpdatedThemes = 'flash.updated-themes',
  UpdatedKeyboardShortCuts = 'flash.keyboard-shortcut-updated',
  UpdatedQunicyEmailSubscription = 'flash.subscribe-to-quincy-updated',
  UpdatedPortfolio = 'flash.portfolio-item-updated',
  UpdatedPrivacy = 'flash.privacy-updated',
  UsernameNotFound = 'flash.username-not-found',
  UsernameTaken = 'flash.username-taken',
  UsernameUpdated = 'flash.username-updated',
  UsernameUsed = 'flash.username-used',
  UserNotCertified = 'flash.user-not-certified',
  WrongName = 'flash.wrong-name',
  WrongUpdating = 'flash.wrong-updating',
  WentWrong = 'flash.went-wrong'
}
