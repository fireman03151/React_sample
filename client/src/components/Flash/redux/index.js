import { createAction, handleActions } from 'redux-actions';
import nanoId from 'nanoid';

import { createTypes } from '../../../utils/createTypes';

export const ns = 'flash';

const initialState = {
  message: {}
};

const types = createTypes(['createFlashMessage', 'removeFlashMessage'], ns);

export const sagas = [];

export const createFlashMessage = createAction(
  types.createFlashMessage,
  msg => ({ id: nanoId(), ...msg })
);
export const removeFlashMessage = createAction(types.removeFlashMessage);

export const flashMessageSelector = state => state[ns].message;

export const reducer = handleActions(
  {
    [types.createFlashMessage]: (state, { payload }) => ({
      ...state,
      message: payload
    }),
    [types.removeFlashMessage]: state => ({
      ...state,
      message: {}
    })
  },
  initialState
);
