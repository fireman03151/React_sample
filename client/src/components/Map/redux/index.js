import { createAction, handleActions } from 'redux-actions';

import { createTypes } from '../../../../utils/stateManagement';

export const ns = 'curriculumMap';

export const getNS = () => ns;

const initialState = {
  expandedState: {
    superBlock: {},
    block: {}
  }
};

const types = createTypes(
  ['resetExpansion', 'toggleSuperBlock', 'toggleBlock'],
  ns
);

export const resetExpansion = createAction(types.resetExpansion);
export const toggleBlock = createAction(types.toggleBlock);
export const toggleSuperBlock = createAction(types.toggleSuperBlock);

export const makeExpandedSuperBlockSelector = superBlock => state =>
  !!state[ns].expandedState.superBlock[superBlock];
export const makeExpandedBlockSelector = block => state =>
  !!state[ns].expandedState.block[block];

export const reducer = handleActions(
  {
    [types.resetExpansion]: state => ({
      ...state,
      expandedState: {
        superBlock: {},
        block: {}
      }
    }),
    [types.toggleBlock]: (state, { payload }) => ({
      ...state,
      expandedState: {
        ...state.expandedState,
        block: {
          ...state.expandedState.block,
          [payload]: !state.expandedState.block[payload]
        }
      }
    }),
    [types.toggleSuperBlock]: (state, { payload }) => ({
      ...state,
      expandedState: {
        ...state.expandedState,
        superBlock: {
          ...state.expandedState.superBlock,
          [payload]: !state.expandedState.superBlock[payload]
        }
      }
    })
  },
  initialState
);
