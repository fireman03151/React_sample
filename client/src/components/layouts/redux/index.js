import { createAction, handleActions } from 'redux-actions';

import { createTypes } from '../../../utils/createTypes';

export const ns = 'guideNav';

const initialState = {
  displaySideNav: false,
  expandedState: {}
};

const types = createTypes(['toggleExpandedState', 'toggleDisplaySideNav'], ns);

export const toggleExpandedState = createAction(types.toggleExpandedState);
export const toggleDisplaySideNav = createAction(types.toggleDisplaySideNav);

export const reducer = handleActions(
  {
    [types.toggleExpandedState]: (state, { payload }) => ({
      ...state,
      expandedState: {
        ...state.expandedState,
        [payload]: !state.expandedState[payload]
      }
    }),
    [types.toggleDisplaySideNav]: state => ({
      ...state,
      displaySideNav: !state.displaySideNav
    })
  },
  initialState
);
