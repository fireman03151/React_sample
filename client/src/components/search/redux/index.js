import { createAction, handleActions } from 'redux-actions';

import { createTypes } from '../../../utils/create-types';

export const ns = 'search';

const initialState = {
  query: '',
  indexName: 'news',
  isSearchDropdownEnabled: true,
  isSearchBarFocused: false
};

const types = createTypes(
  ['toggleSearchDropdown', 'toggleSearchFocused', 'updateSearchQuery'],
  ns
);

export const toggleSearchDropdown = createAction(types.toggleSearchDropdown);
export const toggleSearchFocused = createAction(types.toggleSearchFocused);
export const updateSearchQuery = createAction(types.updateSearchQuery);

export const isSearchDropdownEnabledSelector = state =>
  state[ns].isSearchDropdownEnabled;
export const isSearchBarFocusedSelector = state => state[ns].isSearchBarFocused;
export const searchQuerySelector = state => state[ns].query;

export const reducer = handleActions(
  {
    [types.toggleSearchDropdown]: (state, { payload }) => ({
      ...state,
      isSearchDropdownEnabled:
        typeof payload === 'boolean' ? payload : !state.isSearchDropdownEnabled
    }),
    [types.toggleSearchFocused]: (state, { payload }) => ({
      ...state,
      isSearchBarFocused: payload
    }),
    [types.updateSearchQuery]: (state, { payload }) => {
      return {
        ...state,
        query: payload
      };
    }
  },
  initialState
);
