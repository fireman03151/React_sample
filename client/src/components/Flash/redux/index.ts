import { nanoid } from 'nanoid';
import store from 'store';
import { FlashState, State } from '../../../redux/types';

export const FlashApp = 'flash';

const initialState = {
  message: {
    id: '',
    type: '',
    message: ''
  }
};

export const sagas = [];

export const flashMessageSelector = (state: State): FlashState['message'] =>
  state[FlashApp].message;

// ACTION DEFINITIONS

enum FlashActionTypes {
  CreateFlashMessage = 'createFlashMessage',
  RemoveFlashMessage = 'removeFlashMessage'
}

export type FlashMessageArg = {
  type: string;
  message: string;
  variables?: Record<string, unknown>;
};

export const createFlashMessage = (
  flash: FlashMessageArg
): ReducerPayload<FlashActionTypes.CreateFlashMessage> => {
  const playSound = store.get('fcc-sound') as boolean | undefined;
  if (playSound) {
    void import('tone').then(tone => {
      if (tone.context.state !== 'running') {
        void tone.context.resume();
      }
      if (flash.message === 'flash.incomplete-steps') {
        const player = new tone.Player(
          'https://campfire-mode.freecodecamp.org/try-again.mp3'
        ).toDestination();
        player.autostart = playSound;
      }
      if (flash.message === 'flash.cert-claim-success') {
        const player = new tone.Player(
          'https://campfire-mode.freecodecamp.org/cert.mp3'
        ).toDestination();
        player.autostart = playSound;
      }
    });
  }
  return {
    type: FlashActionTypes.CreateFlashMessage,
    payload: { ...flash, id: nanoid() }
  };
};

export const removeFlashMessage =
  (): ReducerPayload<FlashActionTypes.RemoveFlashMessage> => ({
    type: FlashActionTypes.RemoveFlashMessage
  });

// REDUCER
type ReducerBase<T> = { type: T };
type ReducerPayload<T extends FlashActionTypes> =
  T extends FlashActionTypes.CreateFlashMessage
    ? ReducerBase<T> & {
        payload: FlashState['message'];
      }
    : ReducerBase<T>;

// Does reducer return FlashState or AppState (whole app)?
export const reducer = (
  state: FlashState = initialState,
  action: ReducerPayload<FlashActionTypes>
): FlashState => {
  switch (action.type) {
    case FlashActionTypes.CreateFlashMessage:
      return { ...state, message: action.payload };
    case FlashActionTypes.RemoveFlashMessage:
      return { ...state, message: initialState.message };
    default:
      return state;
  }
};
