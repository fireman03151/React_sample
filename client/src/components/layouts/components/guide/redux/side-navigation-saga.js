import { takeEvery, put, all } from 'redux-saga/effects';

import { toggleExpandedState } from './';
import { types as appTypes } from '../../../../../redux';

function* expandSideNavOnMountSaga() {
  let guideRegex = /^\/guide\//;
  let onGuide = guideRegex.test(window.location.pathname);
  if (onGuide && typeof window !== 'undefined') {
    const pathMap = window.location.pathname
      .slice(1)
      .split('/')
      .slice(0, -1)
      .reduce((map, current, i, pathArray) => {
        const path =
          i !== 0 ? map[pathArray[i - 1]] + `/${current}` : `/${current}`;
        return {
          ...map,
          [current]: path
        };
      }, {});

    let routes = Object.keys(pathMap).map(key => pathMap[key]);

    yield all(routes.map(route => put(toggleExpandedState(route))));
  }
}

export function createSideNavigationSaga() {
  return [takeEvery(appTypes.appMount, expandSideNavOnMountSaga)];
}
