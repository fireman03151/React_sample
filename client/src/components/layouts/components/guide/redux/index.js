import { createAction, handleActions } from 'redux-actions';

import { createTypes } from '../../../../../utils/createTypes';
import { createSideNavigationSaga } from './side-navigation-saga';

export const ns = 'guideNav';

const initialState = {
  displayMenu: false,
  expandedState: {}
};

const types = createTypes(['toggleExpandedState', 'toggleDisplayMenu'], ns);

export const sagas = [...createSideNavigationSaga(types)];

export const toggleExpandedState = createAction(types.toggleExpandedState);
export const toggleDisplayMenu = createAction(types.toggleDisplayMenu);

export const displayMenuSelector = state => state[ns].displayMenu;
export const expandedStateSelector = state => state[ns].expandedState;

export const reducer = handleActions(
  {
    [types.toggleExpandedState]: (state, { payload }) => ({
      ...state,
      expandedState: {
        ...state.expandedState,
        [payload]: !state.expandedState[payload]
      }
    }),
    [types.toggleDisplayMenu]: state => ({
      ...state,
      displayMenu: !state.displayMenu
    })
  },
  initialState
);
