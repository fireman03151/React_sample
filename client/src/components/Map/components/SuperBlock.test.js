/* global jest, expect */

import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore } from '../../../redux/createStore';

import { SuperBlock } from './SuperBlock';
import mockChallengeNodes from '../../../__mocks__/challenge-nodes';
import mockIntroNodes from '../../../__mocks__/intro-nodes';

function renderWithRedux(ui) {
  return render(<Provider store={createStore()}>{ui}</Provider>);
}

test('<SuperBlock /> not expanded snapshot', () => {
  const props = {
    introNodes: mockIntroNodes,
    isExpanded: false,
    nodes: mockChallengeNodes,
    superBlock: 'Super Block One',
    toggleSuperBlock: () => {}
  };

  const { container } = render(<SuperBlock {...props} />);

  expect(container).toMatchSnapshot('superBlock-not-expanded');
});

test('<SuperBlock /> expanded snapshot', () => {
  const props = {
    introNodes: mockIntroNodes,
    isExpanded: true,
    nodes: mockChallengeNodes,
    superBlock: 'Super Block One',
    toggleSuperBlock: () => {}
  };

  const { container } = renderWithRedux(<SuperBlock {...props} />);

  expect(container).toMatchSnapshot('superBlock-expanded');
});

test('<SuperBlock should handle toggle clicks correctly', () => {
  const toggleSpy = jest.fn();
  const props = {
    introNodes: mockIntroNodes,
    isExpanded: false,
    nodes: mockChallengeNodes,
    superBlock: 'Super Block One',
    toggleSuperBlock: toggleSpy
  };

  const { container, rerender } = renderWithRedux(<SuperBlock {...props} />);

  expect(toggleSpy).not.toHaveBeenCalled();
  expect(container.querySelector('.map-title h4')).toHaveTextContent(
    'Super Block One Certification (300 hours)'
  );
  expect(container.querySelector('ul')).not.toBeInTheDocument();

  fireEvent.click(container.querySelector('.map-title'));

  expect(toggleSpy).toHaveBeenCalledTimes(1);
  expect(toggleSpy).toHaveBeenCalledWith('Super Block One');

  rerender(
    <Provider store={createStore()}>
      <SuperBlock {...props} isExpanded={true} />
    </Provider>
  );

  expect(container.querySelector('ul')).toBeInTheDocument();
});
