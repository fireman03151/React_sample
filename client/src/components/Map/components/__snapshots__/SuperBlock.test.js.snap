// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuperBlock /> expanded snapshot: superBlock-expanded 1`] = `
<div>
  <li
    class="superblock open"
    id="super-block-one"
  >
    <button
      aria-expanded="true"
      class="map-title"
    >
      <svg
        viewBox="0 0 100 100"
        width="25px"
      >
        <polygon
          points="-6.04047,17.1511 81.8903,58.1985 -3.90024,104.196"
          style="stroke: var(--primary-color); fill: var(--primary-color); stroke-width: 1px;"
          transform="matrix(0.999729, 0.023281, -0.023281, 0.999729, 7.39321, -10.0425)"
        />
      </svg>
      <h4>
        Super Block One Certification (300 hours)
      </h4>
    </button>
    <ul>
      <li
        class="block "
      >
        <button
          aria-expanded="false"
          class="map-title"
        >
          <svg
            viewBox="0 0 100 100"
            width="25px"
          >
            <polygon
              points="-6.04047,17.1511 81.8903,58.1985 -3.90024,104.196"
              style="stroke: var(--primary-color); fill: var(--primary-color); stroke-width: 1px;"
              transform="matrix(0.999729, 0.023281, -0.023281, 0.999729, 7.39321, -10.0425)"
            />
          </svg>
          <h4>
            Block A
          </h4>
          <div
            class="map-title-completed"
          >
            <span>
              <span
                class="sr-only"
              >
                icons.not-passed
              </span>
              <svg
                height="50"
                style="height: 15px; margin-right: 10px; width: 15px;"
                viewBox="0 0 200 200"
                width="50"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <title>
                    icons.not-passed
                  </title>
                  <circle
                    cx="100"
                    cy="99"
                    fill="var(--primary-background)"
                    r="95"
                    stroke="var(--primary-color)"
                    stroke-dasharray="null"
                    stroke-linecap="null"
                    stroke-linejoin="null"
                    stroke-width="10"
                  />
                </g>
              </svg>
            </span>
            <span>
              0/2
            </span>
          </div>
        </button>
        <ul />
      </li>
      <li
        class="block "
      >
        <button
          aria-expanded="false"
          class="map-title"
        >
          <svg
            viewBox="0 0 100 100"
            width="25px"
          >
            <polygon
              points="-6.04047,17.1511 81.8903,58.1985 -3.90024,104.196"
              style="stroke: var(--primary-color); fill: var(--primary-color); stroke-width: 1px;"
              transform="matrix(0.999729, 0.023281, -0.023281, 0.999729, 7.39321, -10.0425)"
            />
          </svg>
          <h4>
            Block B
          </h4>
          <div
            class="map-title-completed"
          >
            <span>
              <span
                class="sr-only"
              >
                icons.not-passed
              </span>
              <svg
                height="50"
                style="height: 15px; margin-right: 10px; width: 15px;"
                viewBox="0 0 200 200"
                width="50"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <title>
                    icons.not-passed
                  </title>
                  <circle
                    cx="100"
                    cy="99"
                    fill="var(--primary-background)"
                    r="95"
                    stroke="var(--primary-color)"
                    stroke-dasharray="null"
                    stroke-linecap="null"
                    stroke-linejoin="null"
                    stroke-width="10"
                  />
                </g>
              </svg>
            </span>
            <span>
              0/2
            </span>
          </div>
        </button>
        <ul />
      </li>
      <li
        class="block "
      >
        <button
          aria-expanded="false"
          class="map-title"
        >
          <svg
            viewBox="0 0 100 100"
            width="25px"
          >
            <polygon
              points="-6.04047,17.1511 81.8903,58.1985 -3.90024,104.196"
              style="stroke: var(--primary-color); fill: var(--primary-color); stroke-width: 1px;"
              transform="matrix(0.999729, 0.023281, -0.023281, 0.999729, 7.39321, -10.0425)"
            />
          </svg>
          <h4>
            Block C
          </h4>
          <div
            class="map-title-completed"
          >
            <span>
              <span
                class="sr-only"
              >
                icons.not-passed
              </span>
              <svg
                height="50"
                style="height: 15px; margin-right: 10px; width: 15px;"
                viewBox="0 0 200 200"
                width="50"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <title>
                    icons.not-passed
                  </title>
                  <circle
                    cx="100"
                    cy="99"
                    fill="var(--primary-background)"
                    r="95"
                    stroke="var(--primary-color)"
                    stroke-dasharray="null"
                    stroke-linecap="null"
                    stroke-linejoin="null"
                    stroke-width="10"
                  />
                </g>
              </svg>
            </span>
            <span>
              0/1
            </span>
          </div>
        </button>
        <ul />
      </li>
    </ul>
  </li>
</div>
`;

exports[`<SuperBlock /> not expanded snapshot: superBlock-not-expanded 1`] = `
<div>
  <li
    class="superblock "
    id="super-block-one"
  >
    <button
      aria-expanded="false"
      class="map-title"
    >
      <svg
        viewBox="0 0 100 100"
        width="25px"
      >
        <polygon
          points="-6.04047,17.1511 81.8903,58.1985 -3.90024,104.196"
          style="stroke: var(--primary-color); fill: var(--primary-color); stroke-width: 1px;"
          transform="matrix(0.999729, 0.023281, -0.023281, 0.999729, 7.39321, -10.0425)"
        />
      </svg>
      <h4>
        Super Block One Certification (300 hours)
      </h4>
    </button>
  </li>
</div>
`;
